refactor: Add ALLOW_LIST flags and enforce usage in CheckArgFlags

ryanofsky opened the pull request:
<!-- begin based-on -->
**This is based on #16545.** The non-base commits are:

- [`4a41fdf89a3` common: Add support for combining ArgsManager flags](https://github.com/bitcoin/bitcoin/pull/17580/commits/4a41fdf89a3eb28bc04440f16c2c2f86c2641166)
- [`71ceb477320` scripted-diff: Add ALLOW_LIST flag to arguments retrieved with GetArgs](https://github.com/bitcoin/bitcoin/pull/17580/commits/71ceb477320e86f08ec28a66828e5f401111a7af)
- [`7275db69dd6` refactor: Fix more ALLOW_LIST arguments](https://github.com/bitcoin/bitcoin/pull/17580/commits/7275db69dd6682c9b9eea4fbd87648da4545c404)
- [`84fda7ff060` Always reject empty -blockfilterindex="" arguments](https://github.com/bitcoin/bitcoin/pull/17580/commits/84fda7ff0608f079ad700f7ab3b0697fe3a25448)
- [`d7fb7f07fd1` refactor: Stop calling GetArg/IsArgSet on ALLOW_LIST options](https://github.com/bitcoin/bitcoin/pull/17580/commits/d7fb7f07fd16d33551c79830d0cbe96a7a942693)
- [`da42c7f6671` refactor: Always enforce ALLOW_LIST in CheckArgFlags](https://github.com/bitcoin/bitcoin/pull/17580/commits/da42c7f66715833438a65220eabffe3461884a6d)<!-- end -->

---

Except for two `rpcauth` and `blockfilterindex` fixes (update: rpcauth fix was merged separately in #30401) this PR does not change any behavior outside of tests. It is just supposed to enforce internal consistency and prevent bugs by ensuring that list arguments are always retrieved with GetArgs() and non-list arguments are always retrieved with GetArg(). Followup PRs could use the ALLOW_LIST flags for better documentation and error checking in the future. For example, #17493 builds on this to disallow conflicting config values. 

This change was originally made as part of #17493

ryanofsky committed:
common: Implement ArgsManager flags to parse and validate settings on startup

This commit implements support for new ALLOW_BOOL, ALLOW_INT, ALLOW_STRING, and
ALLOW_LIST flags by validating settings with these flags earlier on startup and
providing detailed error messages to users.

The new flags implement stricter error checking than ALLOW_ANY. For example, a
double negated option like -nosetting=0 is treated like an error instead of
true, and an unrecognized bool value like -setting=true is treated like an
error instead of false. And if a non-list setting is assigned multiple times in
the same section of a configuration file, the later assignments trigger errors
instead of being silently ignored.

The new flags also provide type information that allows ArgsManager
GetSettings() and GetSettingsList() methods to return typed integer and boolean
values instead of unparsed strings.

The changes in this commit have no effect on current application behavior
because the new flags are only used in unit tests. The existing ALLOW_ANY
checks in the argsman_tests/CheckValueTest confirm that no behavior is changing
for current settings, which use ALLOW_ANY.

ryanofsky committed:
scripted-diff: Add ALLOW_LIST flag to arguments retrieved with GetArgs

This change has no effect on behavior, and is basically just a documentation
change at this point. The ALLOW_LIST flag is currently ignored unless
ALLOW_BOOL, ALLOW_INT, or ALLOW_STRING flags are also present, and these flags
are not used yet.

-BEGIN VERIFY SCRIPT-
for f in `git grep -n 'GetArgs(' | grep -v _tests | sed -n 's/.*GetArgs("\([^"]\+\)".*/\1/p' | sort -u`; do
   git grep -l -- "$f" | xargs sed -i "/AddArg(\"$f[=\"]/ s/ArgsManager::ALLOW_ANY/& | ArgsManager::ALLOW_LIST/g"
done
-END VERIFY SCRIPT-

ryanofsky committed:
refactor: Fix more ALLOW_LIST arguments

- Remove ALLOW_LIST flag from bitcoin-wallet -wallet and -debug arguments. They
  are list arguments for bitcoind, but single arguments for bitcoin-wallet.

- Add ALLOW_LIST flag to -includeconf arg (missed by scripted diff since it's
  not accessed through GetArgs)

- Add ALLOW_LIST flag to -proxy, -debug, -loglevel, -whitebind, and -whitelist
  args (missed by scripted diff due to line breaks in AddArgs calls)

- Add ALLOW_LIST flag to -zmq args (missed by scripted diff due to programmatic
  GetArgs calls)

This change has no effect on behavior, and is basically just a documentation
change at this point. The ALLOW_LIST flag is currently ignored unless
ALLOW_BOOL, ALLOW_INT, or ALLOW_STRING flags are also present, and these flags
are not used yet.

hebasto commented:
Concept ACk

promag commented:
Wrong commit bcbbc48a1fe7273494738200d508affef40cb47b?

ajtowns reviewed (commented):
Concept ACK, shame all these checks are happening at runtime rather than compile time though

ryanofsky reviewed (commented):
Thanks for the review!

> shame all these checks are happening at runtime rather than compile time though

Agree, but I think adding checks and types like this PR and #17783 are doing is really the hard part. Switching enforcement from runtime to compile time after constraints are in already place would be easier by comparison

ryanofsky reviewed (commented):
Updated 4f3e5e9988280ac96b7979c352da218bbf939ed0 -> e42d223d75d37de01f68588e4b495d47daee497b ([`pr/wdlist.2`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.2) -> [`pr/wdlist.3`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.3), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.2..pr/wdlist.3)) with suggested checkargflags simplification
Rebased e42d223d75d37de01f68588e4b495d47daee497b -> 9621fae7f267124d7a93409c5320a1113adcbc02 ([`pr/wdlist.3`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.3) -> [`pr/wdlist.4`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.4), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.3-rebase..pr/wdlist.4)) due to conflicts with #18594, #15935, #19561, #19709 on top of #16545 pr/argcheck.18
Rebased 9621fae7f267124d7a93409c5320a1113adcbc02 -> 1fe816edd5ab6a8d495fd6f8f8fda4de4bfeef1a ([`pr/wdlist.4`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.4) -> [`pr/wdlist.5`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.5), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.4-rebase..pr/wdlist.5)) due to conflict with #18267 on top of #16545 pr/argcheck.18
Updated 1fe816edd5ab6a8d495fd6f8f8fda4de4bfeef1a -> 3271ee837547f9c6c1d33866ce6d9efb639a7ad3 ([`pr/wdlist.5`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.5) -> [`pr/wdlist.6`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.6), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.5..pr/wdlist.6)) moving unrelated changes to #17783 on top of #16545 pr/argcheck.18
Rebased 3271ee837547f9c6c1d33866ce6d9efb639a7ad3 -> 678d20b73ded40fe113ecbb6e4df1dc148048aff ([`pr/wdlist.6`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.6) -> [`pr/wdlist.7`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.7), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.6-rebase..pr/wdlist.7)) due to conflicts with #18267, #18309, and #19991 on top of #16545 pr/argcheck.18
Rebased 678d20b73ded40fe113ecbb6e4df1dc148048aff -> f78c85b7bf5cb3c6029451047e8827bfda0fe285 ([`pr/wdlist.7`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.7) -> [`pr/wdlist.8`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.8), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.7-rebase..pr/wdlist.8)) due to conflicts with #19884, #20685, #21060 on top of #16545 pr/argcheck.21
Updated f78c85b7bf5cb3c6029451047e8827bfda0fe285 -> a5626386c0da5cd7dc3d685a2ee0f1488cd454c7 ([`pr/wdlist.8`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.8) -> [`pr/wdlist.9`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.9), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.8..pr/wdlist.9)) fixing silent conflicts and CI errors https://cirrus-ci.com/task/5867084006555648 https://cirrus-ci.com/task/4565262239268864
Rebased a5626386c0da5cd7dc3d685a2ee0f1488cd454c7 -> b83d0e0b35a418207b4c7731ebace9c739716551 ([`pr/wdlist.9`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.9) -> [`pr/wdlist.10`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.10), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.9-rebase..pr/wdlist.10)) on top of #16545 pr/argcheck.22 due to conflicts with #21377, #21710, and #21752
Rebased b83d0e0b35a418207b4c7731ebace9c739716551 -> 601fe2d6f751356836c7bd98441d569c592ea4fe ([`pr/wdlist.10`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.10) -> [`pr/wdlist.11`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.11), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.10-rebase..pr/wdlist.11)) on top of #16545 pr/argcheck.25
Rebased 601fe2d6f751356836c7bd98441d569c592ea4fe -> 71815bc8123c455c5b7cea1bd13534d00e61cde6 ([`pr/wdlist.11`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.11) -> [`pr/wdlist.12`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.12), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.11-rebase..pr/wdlist.12)) on top of #16545 pr/argcheck.28
Rebased 71815bc8123c455c5b7cea1bd13534d00e61cde6 -> f98d2b012afb7890a56ce1a3462b4eac0b5fff3e ([`pr/wdlist.12`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.12) -> [`pr/wdlist.13`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.13), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.12-rebase..pr/wdlist.13)) on top of #16545 pr/argcheck.29 and adding new commit to fix new problem with `-blockfilterindex` arguments
Rebased f98d2b012afb7890a56ce1a3462b4eac0b5fff3e -> 38db444164058d31e0b23ab863e50691de40a20c ([`pr/wdlist.13`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.13) -> [`pr/wdlist.14`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.14), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.13-rebase..pr/wdlist.14)) on top of #16545 pr/argcheck.30 to fix signed integer conversion https://cirrus-ci.com/task/5371023720710144
Rebased 38db444164058d31e0b23ab863e50691de40a20c -> 27685d57acb0604dd670fb6fff9c1e58746cd088 ([`pr/wdlist.14`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.14) -> [`pr/wdlist.15`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.15), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.14-rebase..pr/wdlist.15)) due to conflicts with #22087 and #26489

ryanofsky committed:
common: Update ArgManager GetArg helper methods to work better with ALLOW flags

Update GetArg, GetArgs, GetBoolArg, and GetIntArg helper methods to work
conveniently with ALLOW_BOOL, ALLOW_INT, and ALLOW_STRING flags.

The GetArg methods are convenience wrappers around the GetSetting method. The
GetSetting method returns the originally parsed settings values in their
declared bool/int/string types, while the GetArg wrappers provide extra
type-coercion and default-value fallback features as additional conveniences
for callers.

This commit makes two changes to GetArg, GetArgs, GetBoolArg, and GetIntArg
helper methods when BOOL/INT/STRING flags are used:

1. GetArg methods will now raise errors if they are called with inconsistent
   flags. For example, GetArgs will raise a logic_error if it is called on a
   non-LIST setting, GetIntArg will raise a logic_error if it is called
   on a non-INT setting.

2. GetArg methods will now avoid various type coersion footguns when they are
   called on new BOOL/INT/STRING settings. Existing ALLOW_ANY settings are
   unaffected. For example, negated settings will return "" empty strings
   instead of "0" strings (in the past the "0" strings caused strangeness like
   "-nowallet" options creating wallet files named "0"). The new behaviors are
   fully specified and checked by the `CheckValueTest` unit test.

The ergonomics of the GetArg helper methods are subjective and the behaviors
they implement can be nitpicked and debated endlessly. But behavior of these
helper methods does not dictate application behavior, and they can be bypassed
by calling GetSetting and GetSettingList methods instead. If it's necessary,
behavior of these helper methods can also be changed again in the future.

The changes have no effect on current application behavior because the new
flags are only used in unit tests. The `setting_args` unit test and ALLOW_ANY
checks in the `CheckValueTest` unit test are unchanged and confirm that
`GetArg` methods behave the same as before for ALLOW_ANY flags (returning the
same values and throwing the same exceptions).

ryanofsky committed:
Always reject empty -blockfilterindex="" arguments

Previous behavior was inconsistent: if -blockfilterindex or
-blockfilterindex="" arguments were specified they would normally enable all
block filter indexes, but could also trigger "Unknown -blockfilterindex value"
errors if followed by later -blockfilterindex arguments.

It was confusing that the same -blockfilterindex options could sometime trigger
errors and sometimes not depending on option position. It was also confusing
that an empty -blockfilterindex="" setting could enable all indexes even though
indexes are disabled by default.

New behavior is more straightforward:

- -blockfilterindex and -blockfilterindex=1 always enable indexes
- -noblockfilterindex and -blockfilterindex=0 always disable indexes
- -blockfilterindex="" is always an unknown value error

The meaning of these options no longer changes based on option position.

fanquake commented:
Moved this to draft for now, given it's based on another PR, which itself is a draft.

ryanofsky committed:
test: Add test for settings.json parsing with type flags

The type flags aren't currently used to validate or convert settings in the
settings.json file, but they should be in the future. Add test to check current
behavior that can be extended when flags are applied.

Co-authored-by: Hodlinator <172445034+hodlinator@users.noreply.github.com>

ryanofsky committed:
refactor: Stop calling GetArg/IsArgSet on ALLOW_LIST options

Upcoming commits will make it an error to call GetArg and IsArgSet methods on
list options since these usages are error prone. For example GetArg will return
last command line value but first config value in the list, and IsArgSet will
return true even if the list is empty if the list was negated.

This change is just a refactoring replacing problematic ArgsManager calls with
equivalent calls to avoid changing any behavior. Current behavior could
probably be improved in these cases, but this change should make new problems
less likely to be introduced.

ryanofsky committed:
refactor: Always enforce ALLOW_LIST in CheckArgFlags

Prevent GetArg() from being called on ALLOW_LIST arguments, and GetArgs() from
being called on non-list arguments.

This checking was previously skipped unless typed INT/BOOL/STRING flags were
present, but now it's always done.

This change has no effect on external behavior. It is just supposed to enforce
internal consistency and prevent bugs caused by using the wrong GetArg method
to retrieve settings.

ryanofsky committed:
test: Add tests to demonstrate usage of ArgsManager flags

Co-authored-by: Hodlinator <172445034+hodlinator@users.noreply.github.com>

ryanofsky committed:
doc: Add detailed ArgsManager type flag documention

This commit just adds documentation for the type flags. The flags are actually
implemented in the following two commits.

ryanofsky committed:
common: Add support for combining ArgsManager flags

Let ALLOW_STRING and ALLOW_INT flags be combined with ALLOW_BOOL so string and
int options can be specified without explicit values. This is useful for
imperative settings that trigger new behavior when specified and can accept
optional string or integer values, but do not require them. (For examples, see
the example_options unit test modified in this commit.)

ryanofsky committed:
common: Grammar / formatting tweaks

Implement cleanup suggestions from l0rinc:

https://github.com/bitcoin/bitcoin/pull/16545#discussion_r1773221823
https://github.com/bitcoin/bitcoin/pull/16545#discussion_r1773291395
https://github.com/bitcoin/bitcoin/pull/16545#discussion_r1775897382

ryanofsky commented:
Dropping commit 811670650471b352b5e0755e6f38b85154c2bfb3 which is no longer needed after #30401

Rebased a1d65066621a21ea35c4ec8363cfabb862fb5239 -> 8723b8ed56a03394663945bcdaa00039ccdff5f6 ([`pr/wdlist.19`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.19) -> [`pr/wdlist.20`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.20), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.19-rebase..pr/wdlist.20)) due to various conflicts
Updated 8723b8ed56a03394663945bcdaa00039ccdff5f6 -> d2edef6b1a0bde00a29290229a5964e413c49696 ([`pr/wdlist.20`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.20) -> [`pr/wdlist.21`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.21), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.20..pr/wdlist.21)) adding commit da3bdd5f00b3dcd380b7bd67d2d54ec7c4b9c2bb which previously was part of base PR #16545 but was dropped there to make review easier. This functionality in this commit is needed for the -blockfilterindex argument
Rebased d2edef6b1a0bde00a29290229a5964e413c49696 -> 077e20ffba77ec70b89f12df1a7d47c836fb5a67 ([`pr/wdlist.21`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.21) -> [`pr/wdlist.22`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.22), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.21-rebase..pr/wdlist.22)) due to conflicts with #28358, #31223, and #31483
Rebased 077e20ffba77ec70b89f12df1a7d47c836fb5a67 -> 8f2544b1be1faaf1932b0fe87e4f52c76a8cdaba ([`pr/wdlist.22`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.22) -> [`pr/wdlist.23`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.23), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.22-rebase..pr/wdlist.23)) due to conflicts
Updated 8f2544b1be1faaf1932b0fe87e4f52c76a8cdaba -> b7402367abfc8bd3ee3a21205b96e940223c6cdc ([`pr/wdlist.23`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.23) -> [`pr/wdlist.24`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.24), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.23..pr/wdlist.24)) due to silent conflict with #32425
Updated b7402367abfc8bd3ee3a21205b96e940223c6cdc -> 0c5171ff029d1bf871aef1c5a48863ee7aaf0e3a ([`pr/wdlist.24`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.24) -> [`pr/wdlist.25`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.25), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.24..pr/wdlist.25)) to fix wallet tool test failure cause by bad conflict resolution in push before last
Rebased 0c5171ff029d1bf871aef1c5a48863ee7aaf0e3a -> 973220fe6c55f8bb5c724b0f224c9f1942dde795 ([`pr/wdlist.25`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.25) -> [`pr/wdlist.26`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.26), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.25-rebase..pr/wdlist.26)) to fix for-each-commit merge errors
Updated 973220fe6c55f8bb5c724b0f224c9f1942dde795 -> 14f3c77fef85eedb5abc5297f4761e587acacd6d ([`pr/wdlist.26`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.26) -> [`pr/wdlist.27`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.27), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.26..pr/wdlist.27)) adding another -proxy fix for #17783
<!-- begin push-28 -->
Rebased 14f3c77fef85eedb5abc5297f4761e587acacd6d -> 6e874dee28772b6a9fc1453b029ff2a099d8bfd1 ([`pr/wdlist.27`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.27) -> [`pr/wdlist.28`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.28), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.27-rebase..pr/wdlist.28))<!-- end --> due to for-each-commit merge error https://github.com/bitcoin/bitcoin/actions/runs/16686593625/job/47237155434?pr=17580
<!-- begin push-29 -->
Rebased 6e874dee28772b6a9fc1453b029ff2a099d8bfd1 -> ed3368c06138f2846d7f27012c866c8bfcbc109f ([`pr/wdlist.28`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.28) -> [`pr/wdlist.29`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.29), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.28-rebase..pr/wdlist.29))<!-- end -->
<!-- begin push-30 -->
Rebased ed3368c06138f2846d7f27012c866c8bfcbc109f -> b03447e135e56d9ecc4cc1db0da3bfea81098fae ([`pr/wdlist.29`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.29) -> [`pr/wdlist.30`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.30), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.29-rebase..pr/wdlist.30))<!-- end -->
<!-- begin push-31 -->
Rebased b03447e135e56d9ecc4cc1db0da3bfea81098fae -> 6f5066238047501d3d9d9f8eb2da7f55f6c32dd7 ([`pr/wdlist.30`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.30) -> [`pr/wdlist.31`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.31), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.30-rebase..pr/wdlist.31))<!-- end --> on top of #16545 pr/argcheck.50 
<!-- begin push-32 -->
Rebased 6f5066238047501d3d9d9f8eb2da7f55f6c32dd7 -> d96eb990a42c5e89d156881b417075807b8b38a5 ([`pr/wdlist.31`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.31) -> [`pr/wdlist.32`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.32), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.31-rebase..pr/wdlist.32))<!-- end --> due to conflict with #34267
<!-- begin push-33 -->
Rebased d96eb990a42c5e89d156881b417075807b8b38a5 -> c69f5e7bd4a4f04ae4ed9adf0943ae4ef41f3904 ([`pr/wdlist.32`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.32) -> [`pr/wdlist.33`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.33), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.32-rebase..pr/wdlist.33))<!-- end --> on top of pr/argcheck.54
<!-- begin push-34 -->
Rebased c69f5e7bd4a4f04ae4ed9adf0943ae4ef41f3904 -> 7ff7596b4a5e3ea1ecc54791b046ebe7f865f9f7 ([`pr/wdlist.33`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.33) -> [`pr/wdlist.34`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.34), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.33-rebase..pr/wdlist.34))<!-- end --> on top of #16545 pr/argcheck.54 due to conflict with #35335
<!-- begin push-35 -->
Rebased 7ff7596b4a5e3ea1ecc54791b046ebe7f865f9f7 -> da42c7f66715833438a65220eabffe3461884a6d ([`pr/wdlist.34`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.34) -> [`pr/wdlist.35`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.35), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.34-rebase..pr/wdlist.35))<!-- end --> due to conflict with #35441
<!-- begin push-36 -->
Rebased da42c7f66715833438a65220eabffe3461884a6d -> 581eaaf10fc8ce584332fdcc5f04c245f3df7ed2 ([`pr/wdlist.35`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.35) -> [`pr/wdlist.36`](https://github.com/ryanofsky/bitcoin/commits/pr/wdlist.36), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/wdlist.35-rebase..pr/wdlist.36))<!-- end --> due to conflicts with #35205 and #35729

ryanofsky committed:
Merge branch 'pr/argcheck' into pr/wdlist

ryanofsky committed:
test-each-commit: Increase fetch depth

Needed due to base PR, can be dropped before merge
